### PR TITLE
Fix #23853 - Form "Create New Customer" and Create New Address" at the backend doesn't display Prefix and Suffix Options

### LINKED ISSUE #23853: Form "Create New Customer" and Create New Address" at the backend doesn't display Prefix and Suffix Options
### Preconditions (*)

1. Magento 2.3-develop

### Steps to reproduce (*)

1. Store->Configuration->Customers->Customer Configuration
In "Name and Address Options",SET:  Show Prefix: "required", show suffix: "Required", Prefix Dropdown Options: 'mr;mrs', Suffix Dropdown Options: 'jr;sr'
issue_23853_1
2. Go to Create New Customer form from admin
3. View Suffix and Prefix field

### Expected result (*)

1. Display options you can select. The same as on Storefront
issue_23853_2

### Actual result (*)

1. Just display the text field
issue_23853_3

### PR DESCRIPTION
**Pull request by Bartlomiejsz**
### Description (*)

This is fix for #23853 

### Related Pull Requests
https://github.com/magento/inventory/pull/2852

### Fixed Issues (if relevant)

1. magento/magento2#23853: Form "Create New Customer" and Create New Address" at the backend doesn't display Prefix and Suffix Options

### Manual testing scenarios (*)

1. Stores -> Configuration -> Customers -> Customer Configuration
2. In "Name and Address Options",SET: Show Prefix: "required", show suffix: "Required", Prefix Dropdown Options: 'mr;mrs', Suffix Dropdown Options: 'jr;sr'
3. Go to Create New Customer form from admin
4. View Suffix and Prefix field

### Contribution checklist (*)
 - [x] Pull request has a meaningful description of its purpose
 - [x] All commits are accompanied by meaningful commit messages
 - [x] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

### CODE REVIEW

**Review by dmytro-ch — CHANGES_REQUESTED** [review]
Hello @Bartlomiejsz, thank you for your contribution. Due to [Magento Definition of Done](https://devdocs.magento.com/guides/v2.3/contributor-guide/contributing_dod.html#automated-tests) all changes should be covered by tests. Could you please cover your fix by automated tests?
Thank you!

**Comment by Bartlomiejsz** [comment]
Hi @dmytro-ch, done!

**Review by dmytro-ch — APPROVED** [review]

**Comment by magento-engcom-team** [comment]
Hi @dmytro-ch, thank you for the review.
[ENGCOM-6156](https://jira.corp.magento.com/browse/ENGCOM-6156) has been created to process this Pull Request

**Comment by engcom-Alfa** [comment]
:heavy_check_mark: QA Passed

**Comment by slavvka** [comment]
@magento run all tests

**Comment by slavvka** [comment]
@engcom-Foxtrot please fix Static tests

**Review by krzksz — CHANGES_REQUESTED** [review]
@Bartlomiejsz can you please take a look at failing static tests?

**Review by krzksz — APPROVED** [review]

**Comment by magento-engcom-team** [comment]
Hi @krzksz, thank you for the review.
[ENGCOM-6156](https://jira.corp.magento.com/browse/ENGCOM-6156) has been created to process this Pull Request

**Comment by SeruyV** [comment]
Hi @Bartlomiejsz , @Nazar65 please check this case:

- create an additional website (Website2)
- set different values for prefix and suffix at different websites:
 **default config:**
 Selection_007
 **website2:**
 Selection_006
- start creating a customer (backend)
- select "Associate to Website" for just created a website(Website2)
-  ------ Data should change according to the website. 

Selection_004

Not fixed ;(

**Comment by Bartlomiejsz** [comment]
Hi @SeruyV, working on fix for that

**Comment by Bartlomiejsz** [comment]
@magento give me test instance

**Comment by magento-engcom-team** [comment]
Hi @Bartlomiejsz. Thank you for your request. I'm working on Magento instance for you

**Comment by magento-engcom-team** [comment]
Hi @Bartlomiejsz, here is your [new Magento instance](https://pr-25159.instances.magento-community.engineering).
Admin access: [https://pr-25159.instances.magento-community.engineering/admin_0c38](https://pr-25159.instances.magento-community.engineering/admin_0c38)
Login: `d8f985fd` Password: `9050f63f7125`
Instance will be terminated in up to 3 hours.

**Comment by Bartlomiejsz** [comment]
I don't think functional tests are connected to this PR and in static tests I haven't found anything to be fixed.
So @dmytro-ch  @krzksz could you please review again with latest changes?

**Comment by slavvka** [comment]
@Bartlomiejsz Functional tests are probably flaky. I restarted them to make them pass. In Static Tests I can see a JS failure in Static B2B.

**Comment by Bartlomiejsz** [comment]
Hi @slavvka, right, I fixed it, static tests passed :)

File: app/code/Magento/Customer/Model/AttributeMetadataResolver.php [hunk 1]
```diff
@@ -75,25 +78,34 @@ class AttributeMetadataResolver
...
-        ShareConfig $shareConfig
+        ShareConfig $shareConfig,
+        ?PrefixSuffixWithWebsites $prefixSuffixWithWebsites = null
     ) {
         $this->countryWithWebsiteSource = $countryWithWebsiteSource;
         $this->eavValidationRules = $eavValidationRules;
         $this->fileUploaderDataResolver = $fileUploaderDataResolver;
         $this->context = $context;
         $this->shareConfig = $shareConfig;
+        $this->prefixSuffixWithWebsites = $prefixSuffixWithWebsites ?: ObjectManager::getInstance()
```
Comment (slavvka) [inline app/code/Magento/Customer/Model/AttributeMetadataResolver.php]: For 2.4.0 you can inject dependencies naturally ;)

Comment (Bartlomiejsz) [reply app/code/Magento/Customer/Model/AttributeMetadataResolver.php]: Done @slavvka :)

**Comment by slavvka** [comment]
@Bartlomiejsz Could you please verify the failing tests if they are caused by your changes?

**Comment by Bartlomiejsz** [comment]
@slavvka sure, I'll take a look

**Comment by Bartlomiejsz** [comment]
@slavvka looks like there is some issue indeed in EE version caused by those changes, I will have to dig into that

**Comment by sivaschenko** [comment]
@magento run all tests

**Comment by lenaorobei** [comment]
Hi @Bartlomiejsz. Could you please pull latest changes from `2.4-develop` and resolve conflict? Thanks you.

**Comment by Bartlomiejsz** [comment]
Hi @lenaorobei, done, could you please re-run tests?

**Comment by Bartlomiejsz** [comment]
@magento run all tests

**Comment by Bartlomiejsz** [comment]
@magento run all tests

**Comment by lbajsarowicz** [comment]
@magento run all tests
Nah, reported the issue

**Comment by engcom-Echo** [comment]
@magento run all tests

**Comment by Bartlomiejsz** [comment]
@magento run all tests

**Comment by engcom-Hotel** [comment]
Hello @Bartlomiejsz,

Thanks for your contribution!

In order to move further with this PR, I request you to please resolve the conflicts.

Meanwhile, we are moving this PR `On Hold`.

Thanks

**Comment by engcom-Hotel** [comment]
Hello @Bartlomiejsz,

As per this comment https://github.com/magento/magento2/pull/25159#issuecomment-1683511687, we were unable to proceed further because of the conflicts.

For now, we are closing this PR, please feel free to reopen it and resolve the conflicts. We will be happy to process this PR further.

Thanks

**Pull request closed without merging**